packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts

```diff
@@ -0,0 +1,85 @@
+ */
+export async function validateService(
+    url: string,
+    { odataService, axiosConfig }: { odataService: ODataService; axiosConfig: AxiosRequestConfig },
+    requiredVersion: OdataVersion | undefined = undefined,
+    ignoreCertError = false
+): Promise<boolean | string> {
+    try {
+        if (ignoreCertError === true) {
+            ConnectionValidator.setGlobalRejectUnauthorized(!ignoreCertError);
+        }
+        const metadata = await odataService.metadata();
+        const serviceOdataVersion = parseOdataVersion(metadata);
+
+        if (requiredVersion && requiredVersion !== serviceOdataVersion) {
+            return `${t('errors.odataServiceVersionMismatch', {
+                serviceVersion: serviceOdataVersion,
+                requiredVersion
+            })}`;
+        }
+
+        // Remove all occurrences of the origin from the metadata to make backend uris relative
+        PromptState.odataService.metadata = originToRelative(metadata);
+        PromptState.odataService.odataVersion = serviceOdataVersion;
+
+        // Extract sap-client and keep the rest of the query params as part of the url
+        const fullUrl = new URL(url);
+        const sapClient = fullUrl.searchParams.get(SAP_CLIENT_KEY) ?? undefined;
+        fullUrl.searchParams.delete(SAP_CLIENT_KEY);
+
+        PromptState.odataService.servicePath = `${fullUrl.pathname}${fullUrl.search}`;
+        PromptState.odataService.origin = fullUrl.origin;
+        PromptState.odataService.sapClient = sapClient;
+
+        // Best effort attempt to get annotations but dont throw an error if it fails as this may not even be an Abap system
+        try {
+            // Create an abap provider instance to get the annotations using the same request config
+            const abapProvider = createForAbap(axiosConfig);
+            const catalogService = abapProvider.catalog(serviceOdataVersion as unknown as ODataVersion);
+            LoggerHelper.attachAxiosLogger(catalogService.interceptors);
+            LoggerHelper.logger.debug('Getting annotations for service');
+            const annotations = await catalogService.getAnnotations({ path: fullUrl.pathname });
+            LoggerHelper.logger.debug(`Annotations array of length: ${annotations?.length} returned`);
+
+            if (annotations?.length === 0 || !annotations) {
+                LoggerHelper.logger.info(t('prompts.validationMessages.annotationsNotFound'));
+            }
+            PromptState.odataService.annotations = annotations;
+        } catch (err) {
+            LoggerHelper.logger.info(t('prompts.validationMessages.annotationsNotFound'));
+        }
+        return true;
+    } catch (error) {
+        delete PromptState.odataService.metadata;
+        // Provide a more specific error message if the metadata service URL is not found
+        if (ErrorHandler.getErrorType(error) === ERROR_TYPE.NOT_FOUND) {
+            // No metdata implies not a valid odata service
```

Comment:
typo

Reply:
Thx! Will fix is pending PR: https://github.com/SAP/open-ux-tools/pull/2036